winml/lib/Api/LearningModelDevice.cpp

```diff
@@ -129,9 +129,79 @@ LearningModelDevice::CacheThreadPool(_winml::IThreading* thread_pool) {
   return S_OK;
 }
 
+struct LogicalProcessorInformation {
```

Comment:
This change is great. Can we move the code to a common place so that the other build variants can use the same default setting?

Reply:
Ill look into putting the core counting code in a more central location.

Reply:
> This is windows only though

I mean all ORT Windows build can use this logic, regardless which EPs are enabled. The CPU EP can always use this logic.

Reply:
Well this is for calculating default thread count, which I believe is a WinML thing?

Reply:
The core framework has some similar code:
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/util/thread_utils.cc#L78
It also needs to calculate the default thread count, which should be the number of physical cores(if we don't consider CPUs with SoC cores).

Reply:
Will look into making changes in ORT in a future PR which to unblock the changes partners are looking for in an immediate timeframe.

Reply:
@smk2007, could you please move it to the core framework as Changming mentioned above? CPU EP hits same issue:https://github.com/microsoft/onnxruntime/issues/18590